src/main/java/com/sleekbyte/tailor/utils/Finder.java

```diff
@@ -0,0 +1,90 @@
+package com.sleekbyte.tailor.utils;
+
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * Finds all files that match the specified pattern.
+ */
+public final class Finder extends SimpleFileVisitor<Path> {
+
+    private Set<String> includeSet = new HashSet<>();
+    private Set<String> excludeSet = new HashSet<>();
+    private Set<String> fileNames = new TreeSet<>();
+
+    public Finder(Set<String> includeSet, Set<String> excludeSet) {
+        this.includeSet = includeSet;
+        this.excludeSet = excludeSet;
+    }
+
+    public Set<String> getFileNames() {
+        return fileNames;
+    }
+
+    // Compares the glob pattern against the file or directory name.
+    boolean entityOfInterest(Path file) {
+        Path name = file.getFileName();
+
+        if (name == null) {
+            return false;
+        }
+
+        // If file under inspection is a blacklisted file then return false
+        for (String pattern : excludeSet) {
+            if (FileSystems.getDefault().getPathMatcher("glob:" + pattern).matches(name)) {
```

Comment:
Yes, if `one/two` is excluded, then its entire subtree is not traversed (line 78).